### PR TITLE
Improve Documentation of Project Architecture

### PR DESCRIPTION
**Pull request by PeterNerlich**
### Short description
This PR aims to allow new-hires like myself and new volunteers to quickly get a sufficient understanding of the project architecture.

### Proposed changes

- The *Reference* section in the documentation index is moved up to be after *First Steps* and before *Basic Concepts* and the following sections, which describe the technical requirements of the project.
- The *Reference* section is being renamed to *Project Architecture / Reference* to better signal it as the starting point for understanding the project.
- The descriptions of each django app have been rewritten to have more concise wording and provide sufficient explanation to get a good first idea about what each app does.
- The [wiki for the municipalities](https://wiki.integreat-app.de/) is linked to allow the reader to further familiarise themselves with the perspective of the backend user and get a better understanding where each component fits into.
- The *Reference* is now also in the README.md
- The README.md now includes a hint on how to generate the documentation yourself.

### Side effects

Since this only affects content in the documentation and README, no side effects are expected.

### Resolved issues

*None*

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by ulliholtgrave — APPROVED** [review]
Looks good to me 👍

**Review by MizukiTemma — APPROVED** [review]
Cool! Thank you 😃 

Can you squash your commits before merge?

File: README.md [hunk 1]
```diff
@@ -49,6 +49,25 @@ For detailed instructions, tutorials and the source code reference have a look a
...
+	- API: This app provides wrapper functions around all API routes and classes mapping the cms models to API JSON responses.
+	- CMS: This app contains all database models, views, forms and templates forming the content management system for backend users.
+	- Core: This is the project’s main app which contains all configuration files.
+	- Firebase API: This app provides wrapper functions around the Firebase API to send push notifications.
+	- Gvz API: This app provides wrapper functions around our Gemeindeverzeichnis API to automatically import coordinates and region aliases.
+	- Nominatim API: This app provides wrapper functions around our Nominatim API to automatically import region bounding boxes.
+	- Sitemap: This app dynamically generates a sitemap.xml for the webapp.
+	- SUMM.AI API: This app provides wrapper functions around the SUMM.AI API for automatic translations into Easy German.
+	- XLIFF: This app allows (de-)serialization of translations from/to XLIFF (XML Localization Interchange File Format) for standardised exchange with translation agencies.
+- Tests: This app contains all tests to verify integreat-cms works as intended
```
Comment (timoludwig) [inline README.md:67]:
```suggestion
- [Integreat CMS](integreat_cms): The main package of the integreat-cms with the following sub-packages:
	- [API](integreat_cms/api): This app provides wrapper functions around all API routes and classes mapping the cms models to API JSON responses.
	- [CMS](integreat_cms/cms): This app contains all database models, views, forms and templates forming the content management system for backend users.
	- [Core](integreat_cms/core): This is the project’s main app which contains all configuration files.
	- [Firebase API](firebase_api): This app provides wrapper functions around the Firebase API to send push notifications.
	- [GVZ API](integreat_cms/gvz_api): This app provides wrapper functions around our Gemeindeverzeichnis API to automatically import coordinates and region aliases.
	- [Nominatim API](nominatim_api): This app provides wrapper functions around our Nominatim API to automatically import region bounding boxes.
	- [Sitemap](integreat_cms/sitemap): This app dynamically generates a sitemap.xml for the webapp.
	- [SUMM.AI API](integreat_cms/summ_ai_api): This app provides wrapper functions around the SUMM.AI API for automatic translations into Easy German.
	- [XLIFF](integreat_cms/xliff): This app allows (de-)serialization of translations from/to XLIFF (XML Localization Interchange File Format) for standardised exchange with translation agencies.
- [Tests](tests): This app contains all tests to verify integreat-cms works as intended
```
What do you think about adding relative links here?

**Comment by Gaston69** [comment]
:tada:
